### PR TITLE
style: Migrate globals and styles from O2 

### PR DESCRIPTION
**Pull request by milesj**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

- Adds wrapping logic/capabilities for text and tables.
- Adds more global styles that are extremely useful.
- Support locale aware fonts.

## Motivation and Context

Feature parity with our old O2 system.

## Testing

Checked against the style guide.

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/core/src/components/Table/Cell.tsx [hunk 1]
```diff
@@ -14,6 +14,8 @@ export type Props = {
   startAlign?: boolean;
   /** Truncate text and display all on hover. */
   truncate?: boolean;
+  /** Wrap text and white space. */
+  wrap?: boolean;
```
Comment (stefhatcher) [inline packages/core/src/components/Table/Cell.tsx]: aaah - okay! i was curious if `wrap` was a thing (when looking at your other PR using this)

File: packages/core/src/themes/global.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { FontFace, GlobalSheet } from 'aesthetic';
 import { Theme } from '../types';
 
-export default (fontFaces: { [fontFamily: string]: FontFace[] }) => ({ color }: Theme) =>
+export default (fontFaces: { [fontFamily: string]: FontFace[] }) => ({ color, font }: Theme) =>
   ({
     '@global': {
       '*': {
```
Comment (stefhatcher) [inline packages/core/src/themes/global.ts]: would this be a good time to finally add:

```css
:focus:not(:focus-visible) { outline: none }
```

Gets rid of the annoying outline for mouse users but preserves it for keyboard users, and is ignored by browsers that don’t support `:focus-visible`.

File: packages/core/src/components/DatePickerInput/Input/index.tsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ export default class PrivatePickerInput extends DayPickerInput {
     this.input = ref;
 
     if (propagateRef) {
-      propagateRef(ref);
+      propagateRef.current = ref;
```
Comment (milesj) [inline packages/core/src/components/DatePickerInput/Input/index.tsx]: Was breaking storybook.

File: packages/core/src/components/Rating/index.tsx [hunk 4]
```diff
@@ -64,7 +64,7 @@ export class Rating extends React.Component<Props & WithStylesProps> {
               Star = IconStarFull;
             }
 
-            return <Star key={key} color={color} decorative />;
+            return <Star key={key} color={color} size="1.15em" decorative />;
```
Comment (stefhatcher) [inline packages/core/src/components/Rating/index.tsx]: why this change? `Rating` already has size options that were based off of `1em`

File: packages/core/src/themes/global.ts (hunk 2 above)
Comment (stefhatcher) [reply packages/core/src/themes/global.ts]: i see you've added it! resolving this!

File: packages/core/src/components/Rating/index.tsx (hunk 4 above)
Comment (milesj) [reply packages/core/src/components/Rating/index.tsx]: Slightly scaled up to match font size.
